pokeapp/src/components/Sidebar.js

```diff
@@ -1,22 +1,67 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PokeDollarIcon from "../assets/pokeDollar.svg";
 import {
   typeIconMapper,
   abilityMapper,
   gameMapper,
   weaknessIconMapper,
 } from "../utilities/mappers";
-import { useDispatch } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import { buyPokemon, sellPokemon } from "../redux/pokemonSlice";
+import { Typeahead } from "react-bootstrap-typeahead";
+import "react-bootstrap-typeahead/css/Typeahead.css";
+
+const sortAtoZ = (list) => {
+  const sortedAtoZ = list.sort((name1, name2) => {
+    if (name1 < name2) return -1;
+    if (name1 > name2) return 1;
+    return 0;
+  });
+  return sortedAtoZ;
+};
 
 const SideBar = ({
   selectedPokemon,
   makeUpperCase,
   isMoreInfo,
   setIsMoreInfo,
+  setSelectedPokemon,
 }) => {
+  const [showPokemon, setShowPokemon] = useState(false);
+  const [pokeNameList, setPokeNameList] = useState([]);
+
+  // Endpoint. This is the string that will be used by the type ahead
+  const [pokeSearch, setPokeSearch] = useState([]);
+
+  const allPokemon = useSelector((state) => state.pokemon.allPokemon);
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    getPokenames(allPokemon);
+  }, [allPokemon]);
+
+  const getPokenames = (pokeList) => {
+    const names = pokeList.map((pokemon) => {
+      return pokemon.name;
+    });
+    setPokeNameList(sortAtoZ(names));
+  };
+
+  // This event gets triggered by pressing the enter key
+  const enterSubmit = (e) => {
+    if (e.key === "Enter") {
+      updateSidebarPokemon(allPokemon, pokeSearch);
+    }
+  };
+
+  const updateSidebarPokemon = (pokemonList, searchValue) => {
+    for (const pokemon of pokemonList) {
+      if (searchValue[0] === pokemon.name) {
+        setSelectedPokemon(pokemon);
```

Comment:
Let's add a `break;` statement after line 60.  We should do this to stop the for loop from running after discovering the first occurrence of the search string.

Reply:
If we don't it'll potentially call `setSelectedPokemon` for each matching string (if there are any), or it'll keep searching even after it's found a match.